crates/meroctl/src/cli/context/watch.rs

```diff
@@ -69,19 +77,60 @@ impl WatchCommand {
         environment
             .output
             .write(&InfoLine(&format!("Subscribed to context {}", context_id)));
+
+        if let Some(cmd) = &self.exec {
+            environment
+                .output
+                .write(&InfoLine(&format!("Will execute command: {}", cmd)));
+        }
+
         environment
             .output
             .write(&InfoLine("Streaming events (press Ctrl+C to stop):"));
 
+        let mut event_count = 0;
         while let Some(message) = read.next().await {
+            if let Some(max_count) = self.count {
+                if event_count >= max_count {
+                    break;
+                }
+            }
+
             match message {
                 Ok(msg) => {
                     if let WsMessage::Text(text) = msg {
                         let response = serde_json::from_str::<Response>(&text)?;
                         environment.output.write(&response);
+
+                        if let Some(cmd) = &self.exec {
+                            let output = Command::new("sh")
+                                .arg("-c")
+                                .arg(cmd)
```

Comment:
I don't like that this isn't cross-platform, and we shouldn't be using the shell implicitly anyway